frontend/src/container/GridGraphLayout/Graph/utils.ts

```diff
@@ -20,8 +20,7 @@ export const getGraphVisiblityStateOnDataChange = (
 		);
 		const legendFromLocalStore: [
 			{ name: string; dataIndex: LegendEntryProps[] },
-		] = JSON.parse(legendGraphFromLocalStore as string);
-		// let isfound = false;
+		] = JSON.parse(legendGraphFromLocalStore || '[]');
```

Comment:
there is possibility that JSON.parse might throw an error when parsing with dirty data

Reply:
True, will add try-catch.